### PR TITLE
ConfirmNavigation Sample implementation. 

### PR DESCRIPTION
**Pull request by almirvuk**
[[Update] 20 ConfirmNavigation #62](https://github.com/PrismLibrary/Prism-Samples-Forms/issues/62)

I have implemented this ConfirmNavigation sample with Tabbed page, with two tabs and one additional page.  

Page(s): 
* NonAsyncExamplePage
* AsyncExamplePage
* ResultPage

ViewModel(s):
* NonAsyncExamplePageViewModel
* AsyncExamplePageViewModel
* ResultPageViewModel

Explanations:

* **NonAsyncExamplePagePage** is tab page, and it contains `Switch` and `Button` control, `Switch` is binding to `CanNavigateAway` property. `NonAsyncExamplePageViewModel` is implementing `IConfirmNavigation` interface and `CanNavigateAway` is used for returning value in `CanNavigate` method.

* **AsyncExamplePage** is tab page, and it contains `Button`, `Button` and `Command` property is binding to `NavigateAwayCommand`. `AsyncExamplePageViewModel` is implementing `IConfirmNavigationAsync` interface and I am using` IPageDialogService` to ask the user for confirmation to navigate away, the response from `DisplayAlertAsync` is used for the return value of `CanNavigateAsync` method.

* **ResultPage** is only used for navigating to it, and to show some kind of message with `Label` control. It also has `Button` to navigate back (GoBacAsync).

Let me know if there is a need to change and edit something.

P.S. If this example is OK, then I will write more detailed readme.md content when the example code and approach is "confirmed".

### CODE REVIEW

File: 20-ConfirmNavigation/src/PrismSample/Views/MainPage.xaml [hunk 1]
```diff
@@ -1,16 +1,10 @@
...
              x:Class="PrismSample.Views.MainPage">
 
-    <StackLayout>
-        <Label Text="Welcome to Xamarin.Forms!" 
-           HorizontalOptions="Center"
-           VerticalOptions="CenterAndExpand" />
-    </StackLayout>
-
-</ContentPage>
+</TabbedPage>
```
Comment (dansiegel) [inline 20-ConfirmNavigation/src/PrismSample/Views/MainPage.xaml]: There is zero reason to do this if you're creating tabs dynamically. Just register TabbedPage directly

File: 20-ConfirmNavigation/src/PrismSample/Views/ViewAPage.xaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="PrismSample.Views.ViewAPage"
```
Comment (dansiegel) [inline 20-ConfirmNavigation/src/PrismSample/Views/ViewAPage.xaml]: Use a meaningful name for the page

File: 20-ConfirmNavigation/src/PrismSample/Views/ViewBPage.xaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="PrismSample.Views.ViewBPage"
```
Comment (dansiegel) [inline 20-ConfirmNavigation/src/PrismSample/Views/ViewBPage.xaml]: Use a meaningful name for the page

File: 20-ConfirmNavigation/src/PrismSample/Views/ViewCPage.xaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="PrismSample.Views.ViewCPage">
```
Comment (dansiegel) [inline 20-ConfirmNavigation/src/PrismSample/Views/ViewCPage.xaml]: ViewA would be more appropriate here

File: 20-ConfirmNavigation/src/PrismSample/Views/MainPage.xaml (hunk 1 above)
Comment (almirvuk) [reply 20-ConfirmNavigation/src/PrismSample/Views/MainPage.xaml]: You are right, I was not sure if I can remove "MainPage", and did not wanted to use it in another way than to host tabs, but I will remove it.

File: 20-ConfirmNavigation/src/PrismSample/Views/ViewAPage.xaml (hunk 2 above)
Comment (almirvuk) [reply 20-ConfirmNavigation/src/PrismSample/Views/ViewAPage.xaml]: That will not be easy 🤣 but I will change this. Thank you.

**Pull request merged**